### ISSUE TITLE
CyclistSafetySensing

### ISSUE THREAD

**Issue by corntoole** [open-austin/project-ideas#104]
### What problem are we trying to solve?

This project aims to solve the problem of making drivers aware in real-time of near-by cyclists to decrease collisions between drivers and cyclists. Our proposed solution is to embed devices in the environment around intersections to sense nearby cyclists and notify drivers of the cyclist's presence.

### Who will benefit (directly and indirectly) from this project?

Cyclists and drivers will benefit initially. The wider community may benefit if our streets are safer for cyclists and could lead to increase bicycle ridership.

### Where can we find any research/data available/articles?

  - https://svds.com/tensorflow-image-recognition-raspberry-pi/
  - https://www.theguardian.com/cities/2014/apr/22/six-smart-bike-safety-gadgets
  - http://www.phred.org/~josh/bike/SignalDetection.html
  - https://venturebeat.com/2017/06/14/google-open-sources-mobile-first-computer-vision-models-for-tensorflow/
  

### What help is needed at this time?

Initially we need people with backgrounds or interests in machine learning and systems programming (C/C++, python, JavaScript). We may need front-end and back-end web development farther down the road.

#### Phase 1
  - Explore machine learning approaches to image/object recognition
    - computer vision i.e. OpenCV
    - deep learning i.e. TensorFlow
  - Prototype software for recognizing bicyclists
  - Prototype software/firmware to drive flashing light indicators
  - Select a hardware platform from which to build the sensor/indicator. We're currently prototyping with the Raspberry Pi platform.

#### Phase 2
  - Deploy and test device in the field

### What are the next steps (validation, research, coding, design)?

The next step is to collect image data from sources like [ImageNet](http://www.image-net.org/synset?wnid=n02834778) and perhaps live footage from high bike-traffic roads (Smartphone footage is probably enough) and then learn how to run the data through the ML software.

----

#### Checklist for NEW ideas :baby:
Hey, you're official! You're now part of the growing civic hacking community in Austin. Here's a few things to get started (a couple you've probably already done).

- [x] Create this idea issue
- [x] Flesh out the who, where, and what questions above
- [ ] Start the conversation about this new idea on [Slack](https://openaustin-slackin.herokuapp.com/)!
- [ ] Respond with at least one update on this issue within the next month

#### Checklist for ACTIVE projects :fire:
Let's get this project started! When this idea starts taking off, the Projects Core Team will start helping this project's lead(s) out with project management and connecting you to resources you may need. To get there, please complete and check off the following:

- [ ] Complete [Open Leadership 101](https://mozilla.teachable.com/p/open-leadership-101)
- [ ] Create a [GitHub repository](https://github.com/new) and Slack channel for work.
- [ ] Create a [README file](https://mozilla.github.io/open-leadership-training-series/articles/opening-your-project/write-a-great-project-readme/) in your project repository. This file should help newcomers understand what your project is, why it's important, and kinds of help you're looking for.
- [ ] [Create issues](https://mozilla.github.io/open-leadership-training-series/articles/get-your-project-online/project-set-up-for-collaboration-with-github/#assignment--add-your-first-issue) to describe each task that you plan to do or need help with and how a contributor can get started on that task. You might start and stop a lot, so consider issues as your to-do list.

#### Checklist for FEATURED Projects :tada:
To have your project FEATURED on [Open-Austin.org](https://open-austin.org/), complete the following documentation. In past projects, well-documented featured projects have more contributions than other projects.

* [ ] In your README, link to the [Open Austin Community Participation Guidelines](https://docs.google.com/document/d/1OujyBccPpepUXSY5_nP3alzvd81WlCKSeSf_-sU5K-U/edit?usp=sharing) or write your own code of conduct.
* [ ] [Create file: LICENSE](http://choosealicense.com/) to give your project an open license, allowing for sharing and remixing.
* [ ] [Create file: CONTRIBUTING.md](https://github.com/acabunoc/mozsprint-repo-template/blob/master/CONTRIBUTING.md) so others know [how they can contribute](https://mozilla.github.io/open-leadership-training-series/articles/building-communities-of-contributors/write-contributor-guidelines/).

Once all of the above is complete,
- [ ] [Create an issue](https://github.com/open-austin/open-austin.github.io/issues/new) on the open-austin.github.io repo with the title `Add [this project] to projects page. An Open Austin leader will review this issue and post your project :balloon:

If you get stuck at any point, feel free to reach out to the leadership team on Slack by adding @leadership to your message. We're here to help you make real changes to our city.

**Comment by sawillms** [comment]
I demoed OpenPose (https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/master/doc/demo_overview.md) using images of cyclists to Céline (who introduced me to this effort) et al recently and the classifier was spot on. It's quite slow though (~3fps on my Maxwell notebook) presumably due to the additional pose overlay. A YOLO (You Only Look Once) classifier should fit the bill but the classifier in OpenPose is pretrained and good to go for demo purposes. Céline is working on tracking down real intersection footage, all of which can be used for training.

Adding reactive LED signage is straightforward but I think it would have to be placed in the driver's view on the left of the intersection in order to indicate a cyclist is on their right.

**Comment by corntoole** [comment]
When I first read the suggestion of OpenPose, I was concerned about not being able to use it in embedded settings. But I sort of missed the *for demonstration purposes* part. Have you pursued this any further @sawillms ?

**Comment by johnvanlowe** [comment]
https://github.com/ATXCyclistSafetySensing was created during ATX h4C 2018 to continue this effort. Kris is project champion for the project this year and will likely take up task of kicking Open Austin initiative back up.
